my-webxr-app/src/App.tsx

```diff
@@ -34,6 +38,17 @@ const endPoint: number = 1;
 const radius: number = 0.002;
 
 export default function App() {
+    // PCA run check
+    const dataset = new Matrix([
+        [1,2,3,4],
+        [5,5,6,7],
+        [1,4,2,3],
+        [5,3,2,1],
+        [8,1,2,2]
+    ]);
+    console.log("Covariance PCA:",computeCovariancePCA(dataset, 4));
+    console.log("Classic PCA",computeClassicPCA(dataset, 4));
```

Comment:
I don't know how to verify the computed pca results, so I have two different methods and compare them against the other.

Reply:
umm, I don't think it's good to maintain two different methods and have them check each other. Instead of having to test and maintain one method, we now have to test and maintain two methods, plus also make sure we compare each other correctly. 
As long as we have a good range of testing for one method, it should be sufficient to prove that one method works correctly!

Reply:
Yeah, you have a point. I thought of it too and here is why I choose to do otherwise:

1. I don't think there is a good way to test the result PCA, making sure it is correct, in the form of automated unit testing.
2. There are differences in the performance of the two methods so having two we would have a choice of which to use.
3. There could be mathematical exceptions where it could occur in one method but not the other, so leaving two of them here to cross-check is acceptable.

Those questions could be answered by looking into how ml-matrix is implemented and working the maths out, but I think it will be an over-investment for this. Lemme know if something does not sound in my reasons, I love to discuss it further.

Reply:
I see your concerns, @LongQuanNguyen!

If it's quicker to address these points by a discord call, lmk!

1. What about the PCA makes you worried that unit testing or testing a method independently is not enough?

2. If there is one that is faster than the other, we can go with the faster.

3. Do you have some examples of when these mathematical exceptions could happen?

Apart from the maintenance overhead, there are a couple of other reasons why I am opposing the two different functions doing essentially the same thing:
- these can be counted as code duplication unless we decide to use both of them in production
- to maintain or test one method, devs would have to understand not only the being-tested class but the other one as well, increasing the chance of overlooking specific scenarios (especially when these two methods might/might not work with the same data)
- the cross-check tests seem less helpful compared to unit-testing separate methods independently. We expect the result to be some concrete numbers for each test case.
         - For unit-testing separate methods, we can independently test one method by comparing the result with those hard-coded numbers. If the test fails, we know that one method is not working correctly.
         - For those cross-check methods, although we also expect the result to match some concrete numbers, we are comparing the result of one method to the result of another method. If the test fails, we can't be sure which one is problematic from the test. Devs will have to calculate the test case themselves and/or dig into the implementation of both functions to know which one is malfunctioning. So, the cross-check methods aren't very self-validating and make diagnosis more complicated.
 
But I could have been conservative here so you might want to get opinions from other members too!